### DISCUSSION TITLE
My license has a problem on Edge

### DISCUSSION THREAD

**Discussion by espaguetiazul** [aclap-dev/video-downloadhelper#993] [Help and Q&A]
### Your question

I bought a license for Edge and I am using it on Edge, but when I enter it, it says
"License is for Edge but extension build is for Chrome" 
But, I am not using it on Chrome.

Can you help me?
Thank you

### Addon & CoApp version and details

version: 9.0.2.12
target: google
channel: stable
lang: es-419
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"key":"ad5d54a74e4b4a83","mismatch":true,"other_browser":"Edge","this_browser":"Chrome"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0
{
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 12
}
![videodownloader](https://github.com/aclap-dev/video-downloadhelper/assets/175244552/fbec408d-a165-4344-ab6c-4c5e1d5da18f)

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

In the information you posted above, I see these items:

> target: google
> license: {"key":"ad5d54a74e4b4a83","mismatch":true,"other_browser":"Edge","this_browser":"Chrome"}

That says to me you are running Chrome, contrary to your belief you are running Edge.  I can think of a couple of possibilities.

- Your shortcut for Edge has somehow been hijacked & you're running Chrome by accident.
- Your Edge is masquerading as Chrome.

I have a Firefox add-on that allows me to set the `user_agent` of my browser.  This is an attribute that is transmitted from your client browser to whatever web server you are visiting.  It is part of HTML.  This is the mechanism that has your client browser tell the web server what browser you are running.  This add-on lets me masquerade as an older release of Firefox, as Firefox on Linux (which I do not use, I'm running Windows 11), as Firefox on Android (from my desktop), as Chrome, as Edge, as any number of other combinations that aren't true.  It's just a character string that's included in what gets sent from your system when you click a link on a web page.

But I'm not sure that participates in the way VDH detects your system.  I don't know if VDH looks at the `user_agent` when you click the `Copy information to clipboard` button.  All I can tell you is VDH thinks you are using Chrome.  You might want to look more closely at what exactly you are doing.

**Reply by GrampaWildWilly** [reply]
Looking again at your information, I see this:

> UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0

That seems to be VDH recognizing the `user_agent` (that's what UA stands for, as far as I know).  It seems to have the kitchen sink in there, mentioning AppleWebKit, Chrome, Safari, & Edge.  All of them.  I don't pretend to understand all of this but it seems to me it is possible VDH is looking at this `user_agent` string & not drawing the correct conclusion.  Only @paulrouget Paul, the developer, could tell us for certain if there's a bug here.

**Reply by GrampaWildWilly** [reply]
Are you running this in some sort of virtual machine environment?

**Reply by GrampaWildWilly** [reply]
Or . . . Maybe you don't have the right VDH.  The error message implies that you somehow installed the VDH for Chrome extension on your Edge browser.  I didn't think that was even possible.  But that's what your error message seems to be implying.  I'm out of my league here.

**Reply by GrampaWildWilly** [reply]
Instead of waiting for an answer from somebody who actually knows something, you might try this:

1. Uninstall VDH from Edge.
2. Close all your Edge windows.
3. If you have any open instances of Chrome or Firefox, close all of those as well, just to be clean.  I'm guessing this is not applicable but I'm trying to think of everything.
4. Open up Edge again & come back here.  Click the link at the top of any page of this forum labelled Wiki.  That takes you to what is essentially the VDH user manual.
5. On that page, there are colored buttons for each type of VDH.  Click the one for Edge.  This should reinstall VDH & it should be for Edge, since you'll be running Edge & you'll click the button for VDH for Edge.  There should be no confusion in the software at this point.
6. You'll probably have to re-enter your Edge license number.  Let's hope it doesn't lock up.  But there's a solution for that, which I'll point you to below.
7. Do your `Copy information to clipboard` thing again & see if things are straightened out.

For a locked license, take a look at this discussion: #928.  I've got screenshots in there showing what happened when I locked my Firefox license recently.  It was pretty much automatic how I unlocked it.  I'm assuming it works pretty much the same on Edge.

**Comment by espaguetiazul** [comment]
Wooww you are a genius!!! Thank you so much. 
I just follow your last instructions and it works!
Thank you so much

**Comment by espaguetiazul** [comment]
This is whats is says now:


version: 9.0.2.12
target: microsoft
channel: stable
lang: es-419
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"ad5d54a74e4b4a83"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0
{
    "valid_license_message_has_been_displayed": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 2
}

**Reply by GrampaWildWilly** [reply]
Genius?  Maybe, but this sure isn't proof of it.  I just read the details.  But it tickles my fancy to think that I've misled you into thinking that.

So it seems you installed VDH for Chrome on Edge.  How can that even happen?  I assumed that the add-on just wouldn't install if you tried to execute the installer on the wrong browser.  I figured the programming conventions would be such that it would just fail, that code meant for Chrome just wouldn't execute in Edge.  Are the 2 browsers that similar?  I use Firefox so I'm in another world entirely.  Well, this is disturbing.  Maybe this is something worth having @paulrouget Paul look into.  Normally, I would request that you mark my post as the answer.  But I think there's still an open issue here worth Paul's attention.

**Reply by paulrouget** [reply]
Edge allows installing Chrome install. But there's a giant warning when this happens.

**Reply by GrampaWildWilly** [reply]
So my solution was actually the proper one, even though it was just a wild-ass guess on my part.  I guess you can mark my post & close this, then.
